implementation/leprechaun/src/vehicle/controller/vehicle.controller.ts

```diff
@@ -0,0 +1,75 @@
+import { Body, Controller, Delete, Get, HttpCode, Param, ParseUUIDPipe, Patch, Post, Query } from '@nestjs/common';
+import {
+  ApiCreatedResponse,
+  ApiExtraModels,
+  ApiNoContentResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+} from '@nestjs/swagger';
+import { ApiPaginatedResponse } from '../../shared/api/generic-paginated';
+import { PaginatedDto } from '../../shared/api/generic-paginated.dto';
+import { HttpExceptionDto, SchemaMismatchException } from '../../shared/api/http-exceptions';
+import { VehicleService } from '../service/vehicle.service';
+import { CreateVehicleDto, UpdateVehicleDto } from './vehicle-create.dto';
+import { VehicleDto } from './vehicle.dto';
+
+export const UUIDPipe = new ParseUUIDPipe({
+  version: '7',
+  exceptionFactory: () => new SchemaMismatchException(),
+});
```

Comment:
**question:** Does this allow nil UUIDs? And should this be defined here, if we have UUIDs all around?

Reply:
1) It does not. Implementation checks whether it was set as optional (in this case it is not, as its set by default to false) and then does regex match against proper uuid version.
2) I moved it to a different file in `shared` module, as its needed by all modules.